lib/liquid/context.rb

```diff
@@ -12,17 +12,23 @@ module Liquid
   #
   #   context['bob']  #=> nil  class Context
   class Context
-    attr_reader :scopes, :errors, :registers, :environments, :resource_limits
+    attr_reader :scopes, :errors, :registers, :environments, :resource_limits, :static_registers
     attr_accessor :exception_renderer, :template_name, :partial, :global_filter, :strict_variables, :strict_filters
 
-    def initialize(environments = {}, outer_scope = {}, registers = {}, rethrow_errors = false, resource_limits = nil)
+    def self.build(environments: {}, outer_scope: {}, registers: {}, rethrow_errors: false, resource_limits: nil, static_registers: {})
+      new(environments, outer_scope, registers, rethrow_errors, resource_limits, static_registers)
+    end
+
+    def initialize(environments = {}, outer_scope = {}, registers = {}, rethrow_errors = false, resource_limits = nil, static_registers = {})
```

Comment:
Bit of a difficulty here. Lints aren't allowing this because it has too many arguments. To be fair, it does, but I intentionally avoided splitting up this class to much to keep down on allocations.

Thoughts?

Reply:
👍 let's just silence this one.